### PR TITLE
Add lib unit tests

### PR DESCRIPTION
**Pull request by saschagrunert**
This PR adds unit tests to the `lib` package. Beside that I added a config interface and fixed up some possible nil pointer dereferences I found during the test creation.

### CODE REVIEW

**Comment by saschagrunert** [comment]
Mocking of the `runtimeImpl` is not possible any more on the latest master since the interface made private. @sboeuf I know this change was intended and from the current perspective it totally makes sense. Unfortunately only public interfaces can be mocked in go. How do we want to proceed here?

**Comment by sboeuf** [comment]
@saschagrunert oh I completely missed this, I'm truly sorry. I didn't realize this was used for mocking (which is a great way to test IMO), that's why I simply made it private.
Now that there's a use case for this to be public, let's move it back to public then :)

/cc @mrunalp

**Comment by mrunalp** [comment]
Sounds good to me.

**Comment by saschagrunert** [comment]
@mrunralp @sboeuf Alright, I will make it public again. thanks for the fast response! :)

**Comment by sboeuf** [comment]
Great, and sorry for the extra workload here...

File: lib/container_server_test.go [hunk 1]
```diff
@@ -0,0 +1,1066 @@
...
+
+			// When
+			err := sut.LoadSandbox("id")
+
+			// Then
+			Expect(err).NotTo(BeNil())
+		})
+
+		// TODO(sgrunert): uncomment when the following issue is resolved:
+		// https://github.com/kubernetes-sigs/cri-o/issues/1987
```
Comment (umohnani8) [inline lib/container_server_test.go]: #1987 was fixed. This test can be uncommented now.

**Comment by umohnani8** [comment]
Needs rebase @saschagrunert .
Changes LGTM

File: lib/container_server_test.go (hunk 1 above)
Comment (saschagrunert) [reply lib/container_server_test.go]: Ah I verified that it is not resolved yet.

**Comment by haircommander** [comment]
@rhatdan @mrunalp @umohnani8 needs approved label
(is it possible to add approved label if an approver does /lgtm ?)

**Pull request merged**